File: codeforces/1600~1699/1673/a.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  string s;
  cin >> s;
  int n = s.length();
  if (n == 1) {
    cout << "Bob " << s[0] - 'a' + 1 << endl;
    return;
  } else if (n % 2 == 0) {
    int ans = 0;
    for (int i = 0; i < n; ++i) {
      ans += s[i] - 'a' + 1;
    }
    cout << "Alice " << ans << endl;
  } else {
    int total = 0;
    for (int i = 0; i < n; ++i) {
      total += s[i] - 'a' + 1;
    }
    int prefix = total - (s.back() - 'a' + 1);
    int suffix = total - (s[0] - 'a' + 1);
    int alice = max(prefix, suffix);
    int bob = total - alice;
    cout << "Alice " << alice - bob << endl;
  }
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1673/a.rs
use std::io::{stdin, stdout, BufWriter, Write};
use std::str::FromStr;
use std::fmt::Debug;
use std::cmp::max;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_str() -> String {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn solve() {
    let stdout = stdout();
    let mut writer = BufWriter::new(stdout.lock());

    let s = read_str();
    let n = s.len();
    let bytes = s.as_bytes();

    if n == 1 {
        writeln!(writer, "Bob {}", bytes[0] - b'a' + 1).ok();
    } else if n % 2 == 0 {
        let mut ans: i32 = 0;
        for i in 0..n {
            ans += (bytes[i] - b'a' + 1) as i32;
        }
        writeln!(writer, "Alice {}", ans).ok();
    } else {
        let mut total: i32 = 0;
        for i in 0..n {
            total += (bytes[i] - b'a' + 1) as i32;
        }
        let prefix = total - (bytes[n-1] - b'a' + 1) as i32;
        let suffix = total - (bytes[0] - b'a' + 1) as i32;
        let alice = max(prefix, suffix);
        let bob = total - alice;
        writeln!(writer, "Alice {}", alice - bob).ok();
    }
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
